src/components/Menu/UserMenu/index.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import { useWeb3React } from '@web3-react/core'
+import {
+  Flex,
+  LogoutIcon,
+  useModal,
+  UserMenu as UIKitUserMenu,
+  UserMenuDivider,
+  UserMenuItem,
+} from '@pancakeswap/uikit'
+import useAuth from 'hooks/useAuth'
+import { useProfile } from 'state/profile/hooks'
+import ConnectWalletButton from 'components/ConnectWalletButton'
+import { useGetBnbBalance } from 'hooks/useTokenBalance'
+import { useTranslation } from 'contexts/Localization'
+import WalletModal, { WalletView, LOW_BNB_BALANCE } from './WalletModal'
+import ProfileUserMenuItem from './ProfileUserMenutItem'
+import WalletUserMenuItem from './WalletUserMenuItem'
+
+const UserMenu = () => {
+  const { t } = useTranslation()
+  const { account } = useWeb3React()
+  const { logout } = useAuth()
+  const { balance, isFetching } = useGetBnbBalance()
+  const { isInitialized, isLoading, profile } = useProfile()
+  const [onPresentWalletModal] = useModal(<WalletModal initialView={WalletView.WALLET_INFO} />)
+  const [onPresentTransactionModal] = useModal(<WalletModal initialView={WalletView.TRANSACTIONS} />)
```

Comment:
I tried using the 3rd parameter that refreshes props but it just didn't work.

Reply:
Hm.... I was trying to test it out locally but it looks its like one of the situations Hutch was having recently - he wanted to pass props for modal during `onPresentModal` invocation (like `onClick={() => onPresentModal({ props: { initialView: WalletView.TX }}) }`, which is not possible right now. Third prop doesn't really work here cause modal props would need to depend on container props or state, which here is a bit tricky cause we'd need to keep some state for what is about to open and then update it on click and immediately open modal...
So I think the way you did it now is OK for now. Me and Doge were talking about rewriting whole `useModal` hook cause its really annoying to rerender, third prop is more of a temporary fix for exchange modals (and as they just receive their props from other effects its working fine now without extra junk around it). When we got to rewriting useModal we could also make it possible to reset some props during `onPresentModal` so you could declaratively change what is about to open at the point of opening.